ui/src/pages/frontsite/PatronProfile/PatronCurrentLoans/components/ExtendButton/__tests__/ExtendButton.test.js

```diff
@@ -0,0 +1,160 @@
+      availableActions: { extend: 'url/extend' },
+      metadata: {
+        ...testData[0],
+        extension_count: 0,
+      },
+    };
+
+    const component = mount(
+      <ExtendButton
+        extendLoan={() => {}}
+        onExtendSuccess={() => {}}
+        loan={loan}
+        user={user}
+      />
+    );
+    expect(component).toMatchSnapshot();
+  });
+
+  it('should be disabled if it document has pending loans', () => {
+    const loan = {
+      availableActions: { extend: 'url/extend' },
+      metadata: {
+        ...testData[0],
+        extension_count: 0,
+        document: { circulation: { pending_loans: 2 } },
+      },
+    };
+
+    const component = mount(
+      <ExtendButton
+        extendLoan={() => {}}
+        onExtendSuccess={() => {}}
+        loan={loan}
+        user={user}
+      />
+    );
+    expect(component.state().infoMessage).toEqual(
+      'Other users requested that book, therefore you cannot extend your loan.'
+    );
+    const btn = component.find('Button');
+    expect(btn.props().disabled).toBe(true);
+  });
+
+  it('should be disabled if the loan is overdue', () => {
+    const loan = {
+      availableActions: { extend: 'url/extend' },
+      metadata: {
+        ...testData[0],
+        extension_count: 0,
+        is_overdue: true,
+      },
+    };
+
+    const component = mount(
+      <ExtendButton
+        extendLoan={() => {}}
+        onExtendSuccess={() => {}}
+        loan={loan}
+        user={user}
+      />
+    );
+    expect(component.state().infoMessage).toEqual(INFO_MESSAGES.isOverdue);
+    const btn = component.find('Button');
+    expect(btn.props().disabled).toBe(true);
+  });
+
+  it('should be disabled if patron has reached maximum extensions', () => {
+    const loan = {
+      availableActions: { extend: 'url/extend' },
+      metadata: {
+        ...testData[0],
+        extension_count: 3,
+      },
+    };
+
+    const component = mount(
+      <ExtendButton
+        extendLoan={() => {}}
+        onExtendSuccess={() => {}}
+        loan={loan}
+        user={user}
+      />
+    );
+    expect(component.state().infoMessage).toEqual(
+      INFO_MESSAGES.hasMaxExtensions
+    );
+    const btn = component.find('Button');
+    expect(btn.props().disabled).toBe(true);
+  });
+
+  it('should be disabled if patron is not the owner', () => {
+    const loan = {
+      availableActions: { extend: 'url/extend' },
+      metadata: {
+        ...testData[0],
+        extension_count: 3,
+      },
+    };
+    const otherUser = { id: 2 };
+
+    const component = mount(
+      <ExtendButton
+        extendLoan={() => {}}
+        onExtendSuccess={() => {}}
+        loan={loan}
+        user={otherUser}
+      />
+    );
+    expect(component.state().infoMessage).toEqual(INFO_MESSAGES.isOwner);
+    const btn = component.find('Button');
+    expect(btn.props().disabled).toBe(true);
+  });
+
+  it('should be disabled if extend action is in available actions', () => {
```

Comment:
I think you meant to say `should be disabled if extend action is not in available actions` right?